api/unstable/crl.h

```diff
@@ -169,3 +169,69 @@ S2N_API int s2n_crl_lookup_set(struct s2n_crl_lookup *lookup, struct s2n_crl *cr
  * @return S2N_SUCCESS on success, S2N_FAILURE on failure.
  */
 S2N_API int s2n_crl_lookup_ignore(struct s2n_crl_lookup *lookup);
+
+struct s2n_cert_validation_info;
+
+/**
+ * A callback which can be implemented to perform additional validation on received certificates.
+ *
+ * The cert validation callback is invoked after receiving and validating the peer's certificate chain. The callback
+ * can be used by clients to validate server certificates, or by servers to validate client certificates in the case of
+ * mutual auth. Note that any validation performed by applications in the callback is in addition to the certificate
+ * validation already performed by s2n-tls.
+ *
+ * Applications can use either of the following APIs from within the callback to retrieve the peer's certificate chain
+ * and perform validation before proceeding with the handshake:
+ *  - `s2n_connection_get_peer_cert_chain()`
+ *  - `s2n_connection_get_client_cert_chain()`
+ *
+ * If the validation performed in the callback is successful, `s2n_cert_validation_accept()` MUST be called to allow
+ * `s2n_negotiate()` to continue the handshake. If the validation is unsuccessful, `s2n_cert_validation_reject()`
+ * MUST be called, which will cause `s2n_negotiate()` to error. The behavior of `s2n_negotiate()` is undefined if
+ * neither `s2n_cert_validation_accept()` or `s2n_cert_validation_reject()` are called.
```

Comment:
These seems like incorrect behavior. Shouldn't this be an error?

Reply:
If neither accept or reject are called, this is an error:
https://github.com/aws/s2n-tls/blob/e04acc381ae3668d8f4cbcf4ad7ed9dd52bc623a/tls/s2n_x509_validator.c#L626

However, applications are trying to make `s2n_negotiate()` error, which is why they call reject. I wanted to make it clear that not calling accept or reject is not what they should be doing to trigger an error, since we might change this to return blocked if we make it async, so I said the behavior was undefined. Is there a better way to explain this without mentioning the async stuff?